### PR TITLE
Converted IFS browser to TypeScript

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
This PR converts the IFS Browser to TypeScript.

Also:
- Code has been refactored a bit
- Some commands are hidden from the command palette if not intended to be called from there
- IFS Shortcuts and Directory now have an icon
- A French translation has been fixed

### Checklist
* [x] have tested my change
* [x] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Comment by worksofliam** [comment]
@sebjulliand This is so fab! It's even less lines of code - I am going to run with this build for a few days and see how it goes.. will post back if I find any issues.

File: src/views/ifsBrowser.ts [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+    return [];
+  }
+}
+
+class IFSShortcutItem extends IFSDirectoryItem {
+  constructor(readonly shortcut: string) {
+    super({ name: shortcut, path: shortcut, type: "directory" })
+
+    this.contextValue = "shortcut";
+    this.iconPath = new vscode.ThemeIcon("folder-library");
```
Comment (worksofliam) [inline src/views/ifsBrowser.ts:169]: Yes, I love this new icon!

File: src/views/ifsBrowser.ts [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+  sortBy(sort: SortOptions): void {
+    this.ifsParent.sortBy(sort);
+  }
+}
+
+class IFSDirectoryItem extends IFSItem {
+  constructor(file: IFSFile, parent?: IFSDirectoryItem) {
+    super(file, { state: vscode.TreeItemCollapsibleState.Collapsed, parent })
+
+    this.contextValue = "directory";
```
Comment (worksofliam) [inline src/views/ifsBrowser.ts:140]: Directory icon.. so simple yet so nice.

File: src/views/ifsBrowser.ts [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+    this.iconPath = vscode.ThemeIcon.Folder;
+  }
+
+  async getChildren(): Promise<BrowserItem[]> {
```
Comment (worksofliam) [inline src/views/ifsBrowser.ts:144]: Moving the logic to fetch subitems to this right class is a nice tough.

File: src/views/ifsBrowser.ts [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+    super(file, { parent: ifsParent });
+
+    this.contextValue = "streamfile";
+    this.iconPath = vscode.ThemeIcon.File;
+
+    this.resourceUri = vscode.Uri.parse(this.path).with({ scheme: `streamfile` });
+    this.command = {
+      command: `code-for-ibmi.openEditable`,
+      title: `Open Streamfile`,
+      arguments: [file]
```
Comment (worksofliam) [inline src/views/ifsBrowser.ts:127]: It looks like this argument for `openEditable` is wrong. The `file` type is a missmatch to what is expected in `openEditable` (`string`).

image

**Review by worksofliam — CHANGES_REQUESTED** [review]
Most functionality is looking good, other than the little issue that I can't open files. See my comments regarding a type issue.

Other things that are working:

* Download
* Rename / Move
* Search with valid results
* Delete object
* Open in terminal

File: src/views/ifsBrowser.ts (hunk 4 above)
Comment (sebjulliand) [reply src/views/ifsBrowser.ts:127]: Shoot, how did I miss that?!

**Comment by sebjulliand** [comment]
@worksofliam  Thanks for the nice feedback mate!
I fixed the call to `open editable`. It works fine now!

**Review by worksofliam — APPROVED** [review]
@sebjulliand Now we're looking excellent!!

I also testing some additional work to make sure shortcuts are working as expected.

Go ahead and merge whenever you're ready!

**Comment by sebjulliand** [comment]
> @sebjulliand Now we're looking excellent!!
> 
> I also testing some additional work to make sure shortcuts are working as expected.
> 
> Go ahead and merge whenever you're ready!

Lovely! Thanks bud!

**Pull request merged**